consensus/istanbul/backend/engine.go

```diff
@@ -182,6 +182,35 @@ func (sb *backend) VerifyHeader(chain consensus.ChainReader, header *types.Heade
 	return sb.verifyHeader(chain, header, parent)
 }
 
+func getCopiedKIP71ConfigAt(sb *backend, config *params.ChainConfig, blockNum uint64) (*params.ChainConfig, error) {
+	_, data, err := sb.governance.ReadGovernance(blockNum)
+	if err != nil {
+		return nil, err
+	}
+
+	kip71 := params.GetDefaultKIP71Config()
+	gkmr := governance.GovernanceKeyMapReverse
+	if l := data[gkmr[params.LowerBoundBaseFee]]; l != nil {
+		kip71.LowerBoundBaseFee = l.(uint64)
+	}
+	if u := data[gkmr[params.UpperBoundBaseFee]]; u != nil {
+		kip71.UpperBoundBaseFee = u.(uint64)
+	}
+	if g := data[gkmr[params.GasTarget]]; g != nil {
+		kip71.GasTarget = g.(uint64)
+	}
+	if b := data[gkmr[params.BaseFeeDenominator]]; b != nil {
+		kip71.BaseFeeDenominator = b.(uint64)
+	}
+	if m := data[gkmr[params.MaxBlockGasUsedForBaseFee]]; m != nil {
+		kip71.MaxBlockGasUsedForBaseFee = m.(uint64)
+	}
+
+	copiedConfig := config.Copy()
+	copiedConfig.Governance.KIP71 = kip71
```

Comment:
How about this? I think this is simpler than the current code. 
- If KIP71 is not nil, all KIP71 fields should not be nil because the default value of uint64 is 0. 
```
if copiedConfig.Governance != nil && copiedConfig.Governance.KIP71 == nil  {
    copiedConfig.Governance.KIP71 = params.GetDefaultKIP71Config()
```